addons/stock/models/stock_quant.py

```diff
@@ -134,8 +134,9 @@ def _search_on_hand(self, operator, value):
         """Handle the "on_hand" filter, indirectly calling `_get_domain_locations`."""
         if operator not in ['=', '!='] or not isinstance(value, bool):
             raise UserError(_('Operation not supported'))
-        domain_loc = self.env['product.product']._get_domain_locations()[0]
-        quant_ids = [l['id'] for l in self.env['stock.quant'].search_read(domain_loc, ['id'])]
+        domain_loc = self.env['product.product'].with_context(compute_child=False)._get_domain_locations()[0]
+        location_ids = self.env['stock.location'].search([('id', 'child_of', domain_loc[0][2])]).ids
+        quant_ids = [l['id'] for l in self.env['stock.quant'].search_read([('location_id', 'in', location_ids)], ['id'])]
```

Comment:
The search read is useless to get only id, the search is doing the same stuff.
equivalent to:
`quant_ids = self.env['stock.quant'].search([('location_id', 'in', location_ids)]).ids`

Reply:
@ryv-odoo Yeah you're right, I can't believe I missed that 🤦‍♂️